apps/directory/src/functions/bookmarkMapper.js

```diff
@@ -0,0 +1,142 @@
+import router from "../router";
+import { useFiltersStore } from "../stores/filtersStore";
+import { useCollectionStore } from "../stores/collectionStore";
+import { useCheckoutStore } from "../stores/checkoutStore";
+
+function setBookmark(bookmark) {
+  router.push({
+    name: router.currentRoute.name,
+    query: bookmark,
+  });
+}
+
+export async function applyBookmark(watchedQuery) {
```

Comment:
in the molgenis-components (or is it the tables app?) there is similar logic. But I suppose of no use to you

Reply:
Components lib has url encoding in it yes.

Reply:
Not sure why the initial load is slower. but it is way faster that emx1 directory!